tests/upgrades/test_performance_tuning.py

```diff
@@ -0,0 +1,195 @@
+        :param str tune_size: tune size would be medium, large, extra-large,
+        extra-extra-large.
+        """
+        with open('custom-hiera.yaml', 'w') as fd:
+            fd.write("\n".join(DEFAULT_CUSTOM_HIERA_DATA))
+            if mongodb_type:
+                fd.write("\n".join(MMPV1_MONGODB))
+            else:
+                fd.write("\n".join(WIREDTIGER_MONGODB))
+            if tune_size == "medium":
+                fd.write("\n".join(MEDIUM_TUNING_DATA))
+
+    @pre_upgrade
+    def test_pre_custom_hiera_file_validation(self):
+        """ In preupgrade scenario we will validate the default custom hiera file
+        parameter with expected custom_hiera file.
+        """
+        actual_custom_hiera = ssh.command("cat /etc/foreman-installer/custom-hiera.yaml").stdout
+        actual_custom_hiera = set([data.strip() for data in actual_custom_hiera])
+        cmd = 'grep "mongodb::server::storage_engine: \'wiredTiger\'" ' \
+              '/etc/foreman-installer/custom-hiera.yaml'
+        mongodb_type = ssh.command('{}'.format(cmd)).return_code
+        if mongodb_type:
+            expected_custom_hiera = set(DEFAULT_CUSTOM_HIERA_DATA + WIREDTIGER_MONGODB)
+        else:
+            expected_custom_hiera = set(DEFAULT_CUSTOM_HIERA_DATA + MMPV1_MONGODB)
+
+        assert expected_custom_hiera == actual_custom_hiera
+
+    @pre_upgrade
+    def test_pre_performance_tuning_apply(self):
+        """In preupgrade scenario we will apply the medium tuning size.
+
+        :id: preupgrade-83404326-20b7-11ea-a370-48f17f1fc2e1
+
+        :steps:
+            1. Run satellite-installer --disable-system-checks.
+
+        :expectedresults: Medium tuning parameter should be applied.
+
+         """
+        cmd = 'grep "mongodb::server::storage_engine: \'wiredTiger\'" ' \
+              '/etc/foreman-installer/custom-hiera.yaml'
+        mongodb_type = ssh.command('{}'.format(cmd)).return_code
+        self._custom_hiera_file_creation(mongodb_type, "medium")
+
+        ssh.upload_file('custom-hiera.yaml', '/etc/foreman-installer')
+        command_output = ssh.command('satellite-installer --disable-system-checks').stdout
+        command_output = [data.strip(" ") for data in command_output]
+        assert "Success!" in command_output
+
+    @post_upgrade
+    def test_post_performance_tuning_apply(self):
+        """In postupgrade scenario, we will check the tune parameter state after upgrade
+
+        :id: postupgrade-31e26b08-2157-11ea-9223-001a4a1601d8
+
+        :steps:
+            1. Collect the current tuning state from satellite.yaml file.
+            2. Compare it with the expected value.
+            3. If Expected value get match then this scenario would be passed.
+            1. Run satellite-installer --tuning medium --disable-system-checks.
+
+        :expectedresults: Medium tuning parameter should be applied.
+
+         """
+        cmd = 'grep "tuning: medium /etc/foreman-installer/scenarios.d/satellite.yaml'
```

Comment:
Seems you are missing ending double quote here:
```suggestion
        cmd = 'grep "tuning: medium" /etc/foreman-installer/scenarios.d/satellite.yaml'
```

Reply:
Fixed